pkg/storegateway/series_chunks.go

```diff
@@ -381,6 +383,34 @@ func (c *loadingSeriesChunksSetIterator) Next() (retHasNext bool) {
 	}
 
 	nextSet.chunksReleaser = chunksPool
+
+	if c.ignoreNativeHistograms {
+		seriesWriteIdx := 0
+		for _, series := range nextSet.series {
+			chksWriteIdx := 0
+			for _, chk := range series.chks {
+				if chk.Raw == nil {
+					continue
+				}
+				series.chks[chksWriteIdx] = chk
+				chksWriteIdx++
+			}
+			series.chks = series.chks[:chksWriteIdx]
+
+			if len(series.chks) == 0 {
+				continue
+			}
+			nextSet.series[seriesWriteIdx] = series
+			seriesWriteIdx++
+		}
+		nextSet.series = nextSet.series[:seriesWriteIdx]
+
+		if len(nextSet.series) == 0 {
+			nextSet.release()
+			return c.Next()
```

Comment:
With a large number of chunks this can blow the stack memory, please unfold into for loop.

Reply:
each set `(nextSet)` is of 5000 series. Typically very large requests can have hundreds of thousands of series (say 200K). In the worst case where all 200K series have only native histograms, this means that we will recurse with `c.Next()` at most 40 times. Do you think a stack overflow is a threat in this case too?

Reply:
I'll unfold it just to be on the safe side 👍

Reply:
i think this may be overworrying, but unfolding doesn't hurt i guess